pkg/broker/broker.go

```diff
@@ -117,6 +120,128 @@ func (a AnsibleBroker) Bootstrap() (*BootstrapResponse, error) {
 	return &BootstrapResponse{SpecCount: len(specs), ImageCount: imageCount}, nil
 }
 
+func (a AnsibleBroker) Recover() (string, error) {
+	// At startup we should write a key to etcd.
+	// Then in recovery see if that key exists, which means we are restarting
+	// and need to try to recover.
+
+	// do we have any jobs that wre still running?
+	// get all /state/*/jobs/* == in progress
+	// For each job, check the status of each of their containers to update
+	// their status in case any of them finished.
+
+	recoverStatuses, err := a.dao.FindJobStateByState(apb.StateInProgress)
+	if err != nil {
+		// no jobs or states to recover, this is OK.
+		if client.IsKeyNotFound(err) {
+			a.log.Info("No jobs to recover")
+			return "", nil
+		}
+		return "", err
+	}
+
+	a.log.Debug("Recovery refresh login")
+
+	// need a client to refresh the login
+	client, cerr := apb.NewClient(a.log)
+	if cerr != nil {
+		a.log.Error("recovery failed: could not create client during recovery")
+		return "", cerr
+	}
+
+	cerr = client.RefreshLoginToken(a.clusterConfig)
```

Comment:
I don't think we need to make this call at all. The token gets refreshed when we attempt to make any contact with the cluster. Deleting this.